Find_first_missing_positive_integer.rb

```diff
@@ -0,0 +1,26 @@
+=begin 
+This problem was asked by Stripe.
+
+Given an array of integers, find the first missing positive integer in linear time and constant space. 
+In other words, find the lowest positive integer that does not exist in the array. 
+The array can contain duplicates and negative numbers as well.
+
+For example, the input [3, 4, -1, 1] should give 2. The input [1, 2, 0] should give 3.
+
+You can modify the input array in-place.
+=end 
+
+def missingpostiveint(array)
+    array = array.sort 
+    array.each_with_index do |value, index|
+        
+         return (value + 1) if value >= 1 && array[index + 1] != value + 1 

```

Comment:
i think your code will fail for this array input as an example:

```
[-7, -6, -5, -3, -1]
```

The answer to the above should be -4. Which is the smallest number that is missing.

Reply:
Try more examples to find edge cases for your algorithm.